src/compiler/checker.ts

```diff
@@ -10861,8 +10861,9 @@ namespace ts {
             let reportedError = false;
             for (let status = iterator.next(); !status.done; status = iterator.next()) {
                 const { errorNode: prop, innerExpression: next, nameType, errorMessage } = status.value;
-                const sourcePropType = getIndexedAccessType(source, nameType, /*accessNode*/ undefined, errorType);
                 const targetPropType = getIndexedAccessType(target, nameType, /*accessNode*/ undefined, errorType);
+                if (targetPropType === errorType || targetPropType.flags & TypeFlags.IndexedAccess) continue; // Don't elaborate on indexes on generic variables

```

Comment:
Why do you have to look at the `errorType`?

Reply:
`errorType` is what `getIndexedAccessType` returns when the property doesn't exist; ergo we do not want to dive into it. I just moved up the check from a line or two below.